Source/WebGPU/WGSL/EntryPointRewriter.cpp

```diff
@@ -59,6 +59,7 @@ class EntryPointRewriter {
     static AST::TypeDecl& getResolvedType(AST::TypeDecl&);
```

Comment:
from the commit message:

> we check whether a struct is being used as an output of the vertex entry point

Can we just always use a struct for inputs and outputs unconditionally, even if they're empty / only have one member? There's no reason to _not_ just always use them. It would make the compiler's internal logic simpler, at the cost of slightly more characters in the generated output. I think that tradeoff is worth it.

Reply:
Converting non-struct outputs to struct also isn't free (implementation wise)... it requires modifying the AST to 1) declare the struct 2) populate the struct. Maybe I'm just looking at it from the opposite perspective, but I feel like that's more complicated than checking the type of the output, with the side effect of emitting more code. I don't feel very strong about it though, and I'm still understanding the constraints of whether those structs can be used in multiple contexts, etc. so I'm not opposed to revisiting it later.